lua/pl/Map.lua

```diff
@@ -11,13 +11,13 @@
 
 local tablex = require 'pl.tablex'
 local utils = require 'pl.utils'
+local List = require 'pl.List'
```

Comment:
This also fixes error when calling `Map:insert` without previously requiring `pl.List`. Also shows that #144 really needs to be fixed.